src/app/pages/record/record.service.ts

```diff
@@ -0,0 +1,33 @@
+import { Injectable } from '@angular/core';
+import { HttpClient } from '@angular/common/http';
+import { ICategories, IEvents } from '../history/history.interface';
+import { Observable } from 'rxjs';
+
+@Injectable({
+  providedIn: 'root'
+})
+export class RecordService {
+
+  constructor(private http: HttpClient) { }
+
+  public getCategories(): Observable<ICategories[]> {
+    return this.http.get<ICategories[]>('categories');
+  }
+
+  public addCategory(category: ICategories): Observable<ICategories> {
+    return this.http.post<ICategories>('categories', category);
+  }
+
+  public addEvent(event: IEvents): Observable<IEvents> {
+    return this.http.post<IEvents>('events', event);
+  }
+
+  public deleteCategory(id: number): Observable<unknown> {
+    return this.http.delete(`categories/${id}`);
+  }
+
+  public editCategory(id: number | undefined, category: ICategories): Observable<unknown> {
```

Comment:
how is id can be undefined? and why unknown?

Reply:
When I was creating the method in component.ts, I got an error in the console that the id is undefined, so I completed the type. What about unknow type, I forgot add ...put<ICategories>(... in function logic

Reply:
okay let's don't touch this and return to this questions later when we'll be working with this methods